frontend/src/metabase/dashboard/components/DashboardHeader/buttons/DashboardSubscriptionsButton.tsx

```diff
@@ -0,0 +1,30 @@
+import type { ButtonHTMLAttributes } from "react";
+import { t } from "ttag";
+
+import { ToolbarButton } from "metabase/common/components/ToolbarButton";
+import { useHasEmailSetup } from "metabase/common/hooks";
+import { toggleSharing } from "metabase/dashboard/actions";
+import { useDispatch } from "metabase/lib/redux";
+import type { ActionIconProps } from "metabase/ui";
+
+export const DashboardSubscriptionsButton = (
```

Comment:
I created a Linear issue for this as we don't have types for this yet. And types for public-facing components should be added under `embedding-sdk-package` folder instead.